onnxruntime/contrib_ops/cpu/transformers/greedy_search_impl_base.h

```diff
@@ -205,14 +205,16 @@ class GreedySearchBase : public GenerateBase {
 template <typename T, typename ParametersT>
 Status GreedySearchBase<T, ParametersT>::CheckInputs(const OpKernelContextInternal& context) {
   // Input shapes:
-  //   input_ids  : (batch_size, sequence_length)
-  //   vocab_mask : (vocab_size) or nullptr
+  //   input_ids          : (batch_size, sequence_length)
+  //   vocab_mask         : (vocab_size) or nullptr
+  //   decoder_input_ids  : (batch_size, initial_decode_sequence_length)
   ORT_RETURN_IF_ERROR(this->CheckInputsImpl(parameters_,
-                                            context.Input<Tensor>(0),    // input_ids
-                                            context.Input<Tensor>(4),    // vocab_mask
-                                            context.Input<Tensor>(5),    // prefix_vocab_mask
-                                            context.Input<Tensor>(6),    // attention_mask
-                                            context.Input<Tensor>(7)));  // presence_mask
+                                            context.Input<Tensor>(0),     // input_ids
+                                            context.Input<Tensor>(4),     // vocab_mask
+                                            context.Input<Tensor>(5),     // prefix_vocab_mask
+                                            context.Input<Tensor>(6),     // attention_mask
+                                            context.Input<Tensor>(7),     // presence_mask
+                                            context.Input<Tensor>(10)));  // decoder_input_ids
 
```

Comment:
Do we have similar change for GreedySearch?

Reply:
Not currently. This was added to fix an error when building and running. We can look into adding support for greedy search with Whisper.